### ISSUE TITLE
Why is expand-to-line disabled?

### ISSUE THREAD

**Issue by russelldavis** [aronwoost/sublime-expand-region#64]
I was trying to figure out why the selection doesn't expand to the full line, and tracked it down to this commit: https://github.com/aronwoost/sublime-expand-region/commit/eab9e3dc84fb6573a7d0c86b1245573d644f6bb3. Why was that commented out, and would you be open to re-enabling it?

**Comment by r-stein** [comment]
I would assume the reason for this is, that expand to line can break semantical expansions. For example consider the following code (where `|` is the cursor):

```
def foo(bar1, bar2, bar3, |bar4,
        bar5, bar6, bar7):
    ...
```

If we use expand to line this would expand outside the parens, which is not the desired behavior for this case.

However you could checkout and try out https://github.com/aronwoost/sublime-expand-region/pull/59, which solves this issue using sorted closest expansions. I.e. do both expansions and take the expansion, which is inside the other. If they are different take the first one (expand to parens).

I would appreciate feedback and problems for that PR.

**Comment by aronwoost** [comment]
Yes, it was causing trouble with other expansion strategies. Also I considered it not-so-important for the typical use case (= my usage). However, I understand that expand-to-line might be super important for some languages.

What language are you using @russelldavis?

**Comment by russelldavis** [comment]
I'm mainly using Ruby at the moment, but it's problematic for other languages as well. For example, with this line in javascript:

```
var fs = require('fs');
```

Starting on either side of the `=`, the region stops expanding once it hits the `=`.

@r-stein, that sounds like a great solution.

**Comment by aronwoost** [comment]
You can use cmd-L to select line.

**Comment by russelldavis** [comment]
Of course. You can also use cmd-D to select a word. That doesn't mean this plugin should stop selecting words. :)

**Comment by aronwoost** [comment]
Sure, I just wanted to provide an alternative.

It all started with cmd-D, right? :smiley:

**Comment by russelldavis** [comment]
Closing out old issues I've filed that have gone inactive.
